### PR TITLE
add critical test

### PR DESCRIPTION
**Pull request by jianzhangbjz**
as title, 1, replace the `bin-test` with `critical-test` 2, Set the `e2e-test` as optional

### CODE REVIEW

**Comment by jianzhangbjz** [comment]
@bandrade Could you help add `/lgtm` label? Thanks!

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master.yaml [hunk 1]
```diff
@@ -43,9 +43,9 @@ tests:
   steps:
     cluster_profile: aws
     test:
-    - as: bin-test
+    - as: critical-test
       cli: latest
-      commands: ./bin/extended-platform-tests run openshift/conformance/parallel --run "High-24829"
+      commands: ./bin/extended-platform-tests run all --dry-run|grep -E "OLM.*Critical"|./bin/extended-platform-tests run -f -
```
Comment (kuiwang02) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master.yaml:48]: @jianzhangbjz for later usage, please add 
```console
grep -v "VMonly"
```

I am not sure if we need to add 
```console
grep -v "Disruptive"
```
I also suggest to add it, but it is up to you.

Comment (jianzhangbjz) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master.yaml:48]: Thanks!  It's OK to run the `Disruptive` cases there.

**Comment by jianzhangbjz** [comment]
Updated it, @kuiwang02 Could you help add `/lgtm` label? Thanks!

**Pull request merged**